### ISSUE TITLE
Navigation.goBack() animation is not smooth(flashes) when 'reanimated' entering used.

### ISSUE THREAD

**Issue by aubri61** [react-navigation/react-navigation#11305]
### Current behavior

I'm using nested Native Stack Navigator.

And in the inner stack navigators, when i do 'navigation.navigate' the animation works fine.
but when i do 'navigation.goBack()', the screen just flashes and pop the current screen and replace the screen very suddenly.
it seems very unsmooth, and while flashing i can see the white background behind. I really want to fix it since my app's background is dark color..

I found that when i add 'entering' prop to the animated.view from 'react-native-reanimated', this issue happens.

but strangely, when i enable header and press the back button at the header, it works fine(the animation)
look at the video i attached, first one is my current situation. you can see the screen flashing.
but in the second trial, when i press the back button at the header the transition is very smooth.

https://user-images.githubusercontent.com/49435892/227789442-fb32668f-7a31-4ffe-84b5-ecef96348957.mov

i changed component rendering part simply.
and it's not just only problem of this stack. the other stacks also have same problem.

in the snack, it works fine but with the same code, in my app, it does not work... :(

### Expected behavior

when 'navigation.goBack()' the animation for transition performs smoothly.

### Reproduction

https://snack.expo.dev/@aubri61/rn-goback-error

### Platform

- [X] Android
- [X] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [X] @react-navigation/bottom-tabs
- [ ] @react-navigation/drawer
- [ ] @react-navigation/material-top-tabs
- [X] @react-navigation/stack
- [X] @react-navigation/native-stack
- [ ] react-native-tab-view

### Environment

- [] I've removed the packages that I don't use

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native               |"^6.1.2"
| @react-navigation/bottom-tabs          |"^6.5.4",
| @react-navigation/stack                |"^6.3.14",
| @react-navigation/native-stack         |"^6.9.9",
| react-native-safe-area-context         |"^4.5.0",
| react-native-screens                   |"^3.19.0",
| react-native-gesture-handler           |"^2.9.0",
| react-native-reanimated                |"^2.14.4",

| react-native                           |"0.70.0",
| node                                   |
| npm or yarn                            | 9.5.0

**Comment by aubri61** [comment]
i updated those but still same problem :(

**Comment by aubri61** [comment]
and i also changed 'entering' prop into normal animated style and the same problem happen.

`const [tranisitionEnd, setTranisitionEnd] = useState(false);

	useEffect(() => {
		const afterNavigate = navigation.addListener('transitionEnd', () => {
			setTranisitionEnd(true);
		});
		return afterNavigate;
	}, [navigation]);

	const animLogo = useAnimatedStyle(() => {
		const visible = tranisitionEnd ? 1 : 0;

		return {
			opacity: withDelay(
				0,
				withTiming(interpolate(visible, [0, 1], [0, 1]), {
					duration: 1000,
				}),
			),
		};
	});`


`	<Animated.View
				// entering={FadeInDown.duration(1000)}
				style={[animLogo, styles.centerFull, styles.logoCont]}>
				<FastImage
					source={require('@/assets/images/icons/logo/dreamin_logo_11.png')}
					style={styles.logo}
					resizeMode="contain"
				/>
				<Text style={styles.nowText}>
					{'이제 DreamIn을 시작할 수 있습니다.\n당신의 꿈이 당신의 인생이 될 수 있도록.'}
				</Text>
			</Animated.View>`

**Comment by aubri61** [comment]
And there is also issue that if i change 'native stack navigator' into 'stack navigator',

when i come back to the first screen after navigate using goback, the first screen literally dies.
That is, in first screen run 'navigation.navigate('Second')' and then come back to the first using 'navigation.goBack()',
i can't interact with anything on the first screen.
(when the first screen uses reanimated)

if i remove reanimated views, it works fine.
i don't know why it's happening but my App.tsx construct would be problem..?

the order of safe area context and or something.. idk

i'm attaching my App.tsx

```
import React, {useEffect} from 'react';
import {RecoilRoot} from 'recoil';
import {SafeAreaProvider} from 'react-native-safe-area-context';
import {GestureHandlerRootView} from 'react-native-gesture-handler';
import {QueryClientProvider, QueryClient} from 'react-query';

import SplashScreen from 'react-native-splash-screen';

import Navigation from '@/navigation/Navigation';
import theme from '@/styles/theme';

function App(): JSX.Element {
	useEffect(() => {
		setTimeout(() => {
			SplashScreen.hide();
		}, 2000);
	}, []);
	const queryClient = new QueryClient();

	return (
		<SafeAreaProvider>
			<RecoilRoot>
				<QueryClientProvider client={queryClient}>
					<GestureHandlerRootView style={theme.flexView}>
					<Navigation />
					</GestureHandlerRootView>
				</QueryClientProvider>
			</RecoilRoot>
		</SafeAreaProvider>
	);
}

export default App;
```

**Comment by kotano** [comment]
I have the same issue

**Comment by e-younan** [comment]
+1 I've been noticing the same issue but have been too occupied by other things to dive into this. I'm not entirely sure if this is a dev-only issue. Have you tried installing a release version of the application onto a physical device to see if you're still encountering the issue?

**Comment by aubri61** [comment]
@e-younan 
yep! I installed my app into a physical ios device(haven't tried android though) but still got the same problem..

I used react navigation prior version like 2 years ago, but at that time navigation animation works perfectly fine with reanimated.. I guess something went wrong during updating processes maybe..

And I make a note that my friend who also is working on rn project with react navigation and reanimated is suffering from same problems!

Hope this one to be solved soon.. 🙏🏻🥹

**Comment by e-younan** [comment]
In the meantime, you should explicitly set the theme of the NavigationContainer at the root so that you don't see the flicker (if your screens all share the same background colour).

**Comment by ajesamann** [comment]
Also having this issue, and I also installed on a physical device. Both behave exactly like the video shows. Any update on this?

**Comment by mrlmark** [comment]
Ditto

**Comment by cayolegal1** [comment]
> And there is also issue that if i change 'native stack navigator' into 'stack navigator',
> 
> when i come back to the first screen after navigate using goback, the first screen literally dies. That is, in first screen run 'navigation.navigate('Second')' and then come back to the first using 'navigation.goBack()', i can't interact with anything on the first screen. (when the first screen uses reanimated)
> 
> if i remove reanimated views, it works fine. i don't know why it's happening but my App.tsx structure would be problem..?
> 
> the order of safe area context and or something.. idk
> 
> i'm attaching my App.tsx
> 
> ```
> import React, {useEffect} from 'react';
> import {RecoilRoot} from 'recoil';
> import {SafeAreaProvider} from 'react-native-safe-area-context';
> import {GestureHandlerRootView} from 'react-native-gesture-handler';
> import {QueryClientProvider, QueryClient} from 'react-query';
> 
> import SplashScreen from 'react-native-splash-screen';
> 
> import Navigation from '@/navigation/Navigation';
> import theme from '@/styles/theme';
> 
> function App(): JSX.Element {
> 	useEffect(() => {
> 		setTimeout(() => {
> 			SplashScreen.hide();
> 		}, 2000);
> 	}, []);
> 	const queryClient = new QueryClient();
> 
> 	return (
> 		<SafeAreaProvider>
> 			<RecoilRoot>
> 				<QueryClientProvider client={queryClient}>
> 					<GestureHandlerRootView style={theme.flexView}>
> 					<Navigation />
> 					</GestureHandlerRootView>
> 				</QueryClientProvider>
> 			</RecoilRoot>
> 		</SafeAreaProvider>
> 	);
> }
> 
> export default App;
> ```

Same issue using Stack Navigator, moving into Native StackNavigator makes the go back navigation not smoothly

**Comment by BerkeAras** [comment]
Same issue! See here

https://github.com/react-navigation/react-navigation/assets/36629890/d371200e-0b79-446c-b94e-2980302d3c5c
